src/librustc/ty/mod.rs

```diff
@@ -2009,6 +2009,20 @@ impl<'tcx> TyS<'tcx> {
             }
         }
     }
+
+    pub fn has_metadata<'a>(&'tcx self, tcx: TyCtxt<'a, 'tcx, 'tcx>) -> bool {
```

Comment:
The way this is written isn't very "portable", so I'd rather this be moved back to `rustc_trans` (in the only place where it's used).